IO/GeoFileParser.cpp

```diff
@@ -723,56 +723,197 @@ Goal* GeoFileParser::parseGoalNode(TiXmlElement * e)
+     // Read id and check for correct value
+     if(const char* attribute = e->Attribute("id"); attribute) {
+          if(int value = xmltoi(attribute, -1); value > -1 && attribute == std::to_string(value)) {
+               wa->SetId(value);
+          } else {
+               Log->Write("ERROR:\t  waiting area id set but not an integer");
+               delete wa;
+               return nullptr;
+          }
+     } else {
+          Log->Write("ERROR:\t  waiting area id required");
+          delete wa;
+          return nullptr;
+     }
 
-     Log->Write("INFO:\t  Goal id: %d", id);
-     Log->Write("INFO:\t  Goal caption: %s", caption.c_str());
-     Log->Write("INFO:\t  Goal room_id: %d", room_id);
-     Log->Write("INFO:\t  Goal subroom_id: %d", subroom_id);
+     // Read room_id and check for correct value
+     if(const char* attribute = e->Attribute("room_id"); attribute) {
+          if(int value = xmltoi(attribute, -1); value > -1 && attribute == std::to_string(value)) {
+               wa->SetRoomID(value);
+          } else {
+               Log->Write("ERROR:\t  waiting area %d: room_id set but not an integer", wa->GetId());
+               delete wa;
+               return nullptr;
+          }
+     } else {
+          Log->Write("ERROR:\t  waiting area %d: room_id required", wa->GetId());
+          delete wa;
+          return nullptr;
+     }
 
-     Goal* wa = new WaitingArea();
-     WaitingArea* waitingArea = static_cast<WaitingArea*>(wa);
-     waitingArea->SetIsFinalGoal(0);
-     waitingArea->SetId(id);
-     waitingArea->SetCaption(caption);
-     waitingArea->setOpen(open);
-     waitingArea->setGlobalTimer(global_timer);
-     waitingArea->setMinNumPed(min_peds);
-     waitingArea->setMaxNumPed(max_peds);
-     waitingArea->setWaitingTime(waiting_time);
-     waitingArea->setTransitionID(transition_id);
-     waitingArea->SetRoomID(room_id);
-     waitingArea->SetSubRoomID(subroom_id);
+     // Read subroom_id and check for correct value
+     if(const char* attribute = e->Attribute("subroom_id"); attribute) {
+          if(int value = xmltoi(attribute, -1); value > -1 && attribute == std::to_string(value)) {
+               wa->SetSubRoomID(value);
+          } else {
+               Log->Write("ERROR:\t  waiting area %d: subroom_id set but not an integer", wa->GetId());
+               delete wa;
+               return nullptr;
+          }
+     } else {
+          Log->Write("ERROR:\t  waiting area %d: subroom_id required", wa->GetId());
+          delete wa;
+          return nullptr;
+     }
 
-     std::map<int, double> nextGoals;
+     // Read caption and check if correct value
+     if(const char* attribute = e->Attribute("caption"); attribute) {
+          if (std::string value = xmltoa(attribute, ""); !value.empty()) {
+               wa->SetCaption(value);
+          }else{
+               wa->SetCaption("WA " + std::to_string(wa->GetId()));
```

Comment:
Caption is still required in the updated XSD, this code however will be ok with a missing caption

Reply:
Removed the required from the XSD